### PR TITLE
CK Writer 

### PR DESCRIPTION
**Pull request by Kelvinrr**

### CODE REVIEW

File: SugarSpice/src/io.cpp [hunk 1]
```diff
@@ -1,83 +1,197 @@
...
+    this->comment         = comment;
+    this->bodyCode        = bodyCode;
+    this->centerOfMotion  = centerOfMotion;
+    this->referenceFrame  = referenceFrame;
+    this->id              = id;
+    this->polyDegree      = degree;
+    this->statePositions  = statePositions;
+    this->stateVelocities = stateVelocities;
+    this->stateTimes      = stateTimes;
+    return;
```
Comment (tgiroux) [inline SugarSpice/src/io.cpp:30]: Just a style thing but SpkSegment has this empty return and CkSegment does not.

File: SugarSpice/src/io.cpp [hunk 2]
```diff
@@ -1,83 +1,197 @@
...
+             times.size(),
+             times.data(),
+             quats.data(),
+             (angularVelocities) ? angularVelocities->data() : nullptr,
+             times.size(),
+             times.data());
+  }
+
+
+  void writeSpk(fs::path fileName,
```
Comment (tgiroux) [inline SugarSpice/src/io.cpp:99]: just style again, fn argument is called 'path' for writeCk and 'fileName' for writeSpk.

File: SugarSpice/src/io.cpp [hunk 3]
```diff
@@ -1,83 +1,197 @@
...
-    //   if file exists do something (delete it, error out)
-    //   Add convience function to SpkSegment to combine the pos and vel into a single array
 
-    // Combine positions and velocities for spicelib call
-    std::vector<std::vector<double>> pos = segments[0].getStatePositions();
-    std::vector<std::vector<double>> vel = segments[0].getStateVelocities();
-    std::vector<double> times = segments[0].getStateTimes();
+  vector<vector<double>> concatStates (vector<vector<double>> statePositions, vector<vector<double>> stateVelocities) {
+
+    if (statePositions.size() != stateVelocities.size()) {
```
Comment (tgiroux) [inline SugarSpice/src/io.cpp:36]: do they just need to match or is there a constant involved?

File: SugarSpice/tests/IoTests.cpp [hunk 4]
```diff
@@ -1,12 +1,27 @@
...
 
 #include "Fixtures.h"
 #include "io.h"
 
 using namespace SugarSpice;
 
+
+TEST_F(TempTestingFiles, UnitTestWriteCk) {
+  fs::path path;
+  path = "/tmp/test_ck.bsp";
```
Comment (tgiroux) [inline SugarSpice/tests/IoTests.cpp:12]: does this file get removed after the test?

**Review by tgiroux — CHANGES_REQUESTED** [review]
Mostly comments about style plus a couple of questions :+1:

File: SugarSpice/src/io.cpp (hunk 3 above)
Comment (Kelvinrr) [reply SugarSpice/src/io.cpp:36]: You just need to have parallel arrays of velocities and states, so if they are not the same length you cannot make them a Nx6 array

File: SugarSpice/tests/IoTests.cpp (hunk 4 above)
Comment (Kelvinrr) [reply SugarSpice/tests/IoTests.cpp]: I was doing a local test running CKBrief on it, forgot to revert it.

**Review by tgiroux — APPROVED** [review]

**Pull request merged**